src/VecSim/spaces/computer/preprocessor_container.cpp

```diff
@@ -32,15 +31,16 @@ void PreprocessorsContainerAbstract::preprocessQueryInPlace(void *blob,
 void PreprocessorsContainerAbstract::preprocessStorageInPlace(void *blob,
                                                               size_t processed_bytes_count) const {}
 
-MemoryUtils::unique_blob
-PreprocessorsContainerAbstract::maybeCopyToAlignedMem(const void *original_blob,
-                                                      size_t blob_bytes_count) const {
-    if (this->alignment) {
-        if ((uintptr_t)original_blob % this->alignment) {
-            auto aligned_mem = this->allocator->allocate_aligned(blob_bytes_count, this->alignment);
-            memcpy(aligned_mem, original_blob, blob_bytes_count);
-            return this->wrapAllocated(aligned_mem);
-        }
+MemoryUtils::unique_blob PreprocessorsContainerAbstract::maybeCopyToAlignedMem(
+    const void *original_blob, size_t blob_bytes_count, bool force_copy) const {
+    bool needs_copy =
+        force_copy || (this->alignment && ((uintptr_t)original_blob % this->alignment != 0));
+
+    if (needs_copy) {
+        auto aligned_mem = this->allocator->allocate_aligned(blob_bytes_count, this->alignment);
+        // TODO: handle original_blob_size != processed_bytes_count
```

Comment:
When? Is there something that will fail a PR if we forget to support it?

Reply:
ASAP. it is undefined potentially it can lead to a crash